src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -564,6 +676,7 @@ UCS_CLASS_CLEANUP_FUNC(uct_rdmacm_cm_ep_t)
 
     UCS_ASYNC_BLOCK(worker_priv->async);
 
+    uct_rdamcm_cm_ep_destroy_reserved_qpn(self);
     uct_rdmacm_cm_ep_destroy_dummy_qp(self);
```

Comment:
@evgeny-leksikov There are condition checkings in both functions: if (cep->qpn == 0) and if (cep->qp != NULL). This will make sure the code running the right one, and just return directly from the other one. If the the right qp is created, the qp num value won't be 0 or NULL.
This makes the code simpler.

Reply:
@evgeny-leksikov There are condition checkings in both functions: if (cep->qpn == 0) and if (cep->qp != NULL). This will make sure the code running the right one, and just return directly from the other one. If the the right qp is created, the qp num value won't be 0 or NULL.
This makes the code simpler.

Reply:
IMO union is self documented about usage, otherwise please document related fields and their usage

Reply:
also pls incapsulate `uct_rdmacm_cm_ep_destroy_dummy_qp` into `uct_rdamcm_cm_ep_destroy_reserved_qpn` since this qp was created to get qpn

Reply:
@evgeny-leksikov uct_rdamcm_cm_ep_destroy**_reserved**_qpn() is not uct_rdamcm_cm_ep_destroy_qpn(), reserved_qpn is a seperate function. Do you think we should create another function called uct_rdamcm_cm_ep_destroy_qpn() and incapsulate above two functions into it?

Reply:
yes, I think better to abstract this logic in hight level functions if one if/else statement with calling corresponding faunction based on capability

Reply:
@evgeny-leksikov Fixed. No need if/else for destroying since destroying qp type is not decided by the cap, but decided by whether the qpn was created.(Please refer to the comments I added in the code). For creating, I added "one if/else statement to call corresponding function based on capability" as your comments.